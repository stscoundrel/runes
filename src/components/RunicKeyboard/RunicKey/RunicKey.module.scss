.key {
  display: inline-block;
  width: rem(40px);
  height: rem(40px);
  text-align: center;
  font-size: rem(26px);
  line-height: rem(40px);
  letter-spacing: rem(1.5px);
  border: rem(1px) solid $color-dark;
  border-radius: rem(3px);
  color: $color-dark;
  background-color: transparent;
  appearance: none;
  cursor: pointer;

  &:hover,
  &:active,
  &:focus {
    background-color: $color-dark;
    color: white;
  }
}