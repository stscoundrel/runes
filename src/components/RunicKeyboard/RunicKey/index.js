import styles from './RunicKey.module.scss'

export default function RunicKey({ rune, letter, callback }) {
  return (
    <button
      className={styles.key}
      aria-label={`Runic key ${rune}, meaning letter ${letter}`}
      onClick={() => callback(letter)}
    >
      {rune}
    </button>
  )
}
