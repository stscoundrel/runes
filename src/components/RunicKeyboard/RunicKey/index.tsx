import styles from './RunicKey.module.scss'

interface RunicKeyCallback {
  (param: unknown): void
}

interface RunicKeyProps {
  rune: string
  letter: string
  callback: RunicKeyCallback
}

export default function RunicKey({ rune, letter, callback }: RunicKeyProps) {
  return (
    <button
      title={`Runic key ${rune}, meaning letter ${letter}`}
      className={styles.key}
      aria-label={`Runic key ${rune}, meaning letter ${letter}`}
      onClick={() => callback(letter)}
    >
      {rune}
    </button>
  )
}
