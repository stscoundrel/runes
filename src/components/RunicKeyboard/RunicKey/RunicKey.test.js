import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import RunicKey from './index'

describe('RunicKey component', () => {
  test('Does not crash', () => {
    const mockCallBack = jest.fn()
    const div = document.createElement('div')
    ReactDOM.render(<RunicKey rune="ᚠ" letter="f" callback={mockCallBack} />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  test('Matches snapshot', () => {
    const mockCallBack = jest.fn()
    const tree = renderer.create(<RunicKey rune="ᚠ" letter="f" callback={mockCallBack} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Has correct text', () => {
    const mockCallBack = jest.fn()
    const tree = renderer.create(<RunicKey rune="ᚠ" letter="f" callback={mockCallBack} />)
    const { root } = tree

    expect(root.findByType('button').children).toEqual(['ᚠ'])
  })

  test('Triggers callback when clicked', async () => {
    const mockCallBack = jest.fn()
    let tree

    expect(mockCallBack).not.toHaveBeenCalled()

    await renderer.act(async () => {
      tree = renderer.create(<RunicKey rune="ᚠ" letter="f" callback={mockCallBack} />)
    })

    await renderer.act(async () => {
      await tree.root.findByType('button').props.onClick()
    })

    expect(mockCallBack).toHaveBeenCalled()
  })
})
