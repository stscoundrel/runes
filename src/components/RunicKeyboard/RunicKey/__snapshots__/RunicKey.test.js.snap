// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RunicKey component Matches snapshot 1`] = `
<button
  onClick={[Function]}
>
  áš 
</button>
`;
